src/manifest.rs

```diff
@@ -49,7 +55,9 @@ fn read_cargo_toml(path: &str) -> Result<CargoManifest, Error> {
     let mut cargo_contents = String::new();
     cargo_file.read_to_string(&mut cargo_contents)?;
 
-    Ok(toml::from_str(&cargo_contents)?)
+    Ok(toml::from_str(
+        &cargo_contents.replace("crate-type", "crate_type"),
```

Comment:
😱 this makes me nervous. any ideas for a more robust way to handle this?

Reply:
What's the reason you're chaning it from crate-type to crate_type here? It might not be possible if the reason why is valid. If you're deserializing from a file I think we can have serde deserialize the field better with an attribute elsewhere

Reply:
Oh I see now! You'll want to add this attribute to above the `crate_type` field for `CargoLib`

`#[serde(rename = "crate-type")]`

This tells serde that if it sees a field crate-type in the toml file it should be `crate_type` for Rust

Reply:
oooooh nice! very cool